From b62d83bef7733e4943e6f66787166b41198d1359 Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Wed, 13 Apr 2022 16:05:32 +0200
Subject: [PATCH] Dialog: Do not update layout parameters while hidden

Wait until the dialog it's shown to position the components within.
Otherwise we do it based on a random size.

BUG: 452512
---
 src/plasmaquick/dialog.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plasmaquick/dialog.cpp b/src/plasmaquick/dialog.cpp
index 09937ef0f..9ca2c8a66 100644
--- a/src/plasmaquick/dialog.cpp
+++ b/src/plasmaquick/dialog.cpp
@@ -475,7 +475,7 @@ void DialogPrivate::getSizeHints(QSize &min, QSize &max) const
 
 void DialogPrivate::updateLayoutParameters()
 {
-    if (!componentComplete || !mainItem || !mainItemLayout) {
+    if (!componentComplete || !mainItem || !mainItemLayout || q->visibility() == QWindow::Hidden) {
         return;
     }
 
-- 
GitLab

